packages/api/src/controllers/savedSearch.ts

```diff
@@ -13,15 +14,20 @@ export async function getSavedSearches(teamId: string) {
   const alerts = await Alert.find(
     { team: teamId, savedSearch: { $exists: true, $ne: null } },
     { __v: 0 },
-  );
+  ).populate('createdBy', ['email name']);
 
   const alertsBySavedSearchId = groupBy(alerts, 'savedSearch');
 
   return savedSearches.map(savedSearch => ({
     ...savedSearch.toJSON(),
-    alerts: alertsBySavedSearchId[savedSearch._id.toString()]
-      ?.map(alert => alert.toJSON())
-      .map(({ _id, ...alert }) => ({ id: _id, ...alert })), // Remap _id to id
+    alerts: alertsBySavedSearchId[savedSearch._id.toString()]?.map(alert => {
+      const { _id, createdBy, ...restAlert } = alert.toJSON();
+      return {
+        id: _id,
+        ...restAlert,
+        createdBy: createdBy ? pick(createdBy, ['email', 'name']) : undefined,
```

Comment:
then I don't think we need this? since its defined in `populate`?

Reply:
Cleaned up and [added an automated test](https://github.com/hyperdxio/hyperdx/pull/1058/files#diff-3bc82d2285658d336eed82f6edf7cd85dfc4d2108e34caeb3a636d988e58b2a9R117) to check the populate statement.

Reply:
should we keep the id mapping?
```
.map(({ _id, ...alert }) => ({ id: _id, ...alert })), // Remap _id to id
```